src/main/java/ch/uzh/ifi/fs22/sel/group15/cozycave/server/rest/mapper/ListingMapper.java

```diff
@@ -27,18 +29,32 @@ public interface ListingMapper {
     @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
     void updateListingFromListingGetDto(ListingGetDto listingGetDto, @MappingTarget Listing listing);
 
+    @Mapping(source = "publisher", target = "publisher", qualifiedByName = "uuidToUser")
     Listing listingPostDtoToListing(ListingPostDto listingPostDto);
 
+    @Mapping(source = "publisher", target = "publisher", qualifiedByName = "userToUUID")
     ListingPostDto listingPostDto(Listing listing);
 
     // PUT DTO MAPPER
+    @Mapping(source = "publisher", target = "publisher", qualifiedByName = "uuidToUser")
     Listing listingPutDtoToListing(ListingPutDto listingPutDto);
 
+    @Mapping(source = "publisher", target = "publisher", qualifiedByName = "userToUUID")
     ListingPutDto listingToListingPutDto(Listing listing);
 
-    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
+    /*@BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
     void updateListingFromListingPutDto(
-            ListingPutDto listingPutDto, @MappingTarget Listing listing);
+            ListingPutDto listingPutDto, @MappingTarget Listing listing);*/
+
+    @Named("uuidToUser")
+    public static User uuidToUser(UUID uuid) {
+        return new User(uuid, null, null, null, null);
```

Comment:
is using this in the mapper tested? because this looks kinda weird because it only contains the uuid.

Reply:
maybe i look at this again, at the moment i dont have the the time